recipes/mariadb-connector-cpp/all/conanfile.py

```diff
@@ -0,0 +1,146 @@
+
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "with_ssl": [False, "openssl", "gnutls", "schannel"],
+        "with_curl": [True, False]
+        }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_ssl": "openssl",
+        "with_curl": True
+        }
+
+    def export_sources(self):
+        export_conandata_patches(self)
+
+    @property
+    def _min_cppstd(self):
+        return "11"
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def requirements(self):
+        self.requires("mariadb-connector-c/3.3.3")
+        self.requires("zlib/[>=1.2.10 <2]")
+        self.requires("zstd/1.5.5")
+        if self.options.with_ssl == "openssl":
+            self.requires("openssl/[>=1.1 <4]")
+
+    def build_requirements(self):
+        self.tool_requires("cmake/[>=3.23 <4]")
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def validate(self):
+        if self.settings.compiler.get_safe("cppstd"):
+            check_min_cppstd(self, self._min_cppstd)
+        # I dont have a windows computer to test it
+        if self.settings.os == "Windows":
+            raise ConanInvalidConfiguration(f"{self.ref} doesn't support windows for now")
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            self.options.rm_safe("fPIC")
+
+    def configure(self):
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+        # if we are compiling as shared, we need to also compile the mariadbclient as shared
+        self.options["mariadb-connector-c"].shared = self.options.shared
+
+
+    def _package_folder_dep(self, dep):
+        return self.dependencies[dep].package_folder.replace("\\", "/")
+
+    def _include_folder_dep(self, dep):
+        return self.dependencies[dep].cpp_info.includedirs[0].replace("\\", "/")
+
+    def _lib_folder_dep(self, dep):
+        return self.dependencies[dep].cpp_info.libdirs[0].replace("\\", "/")
+
+    def generate(self):
+        env = VirtualBuildEnv(self)
+        env.generate(scope="build")
+        tc = CMakeDeps(self)
+        tc.generate()
+        tc = CMakeToolchain(self)
+        tc.variables["USE_SYSTEM_INSTALLED_LIB"] = True
+        tc.variables["WITH_UNIT_TESTS"] = False
+        tc.variables["BUILD_SHARED_LIBS"] = self.options.shared
+        tc.variables["MARIADB_LINK_DYNAMIC"] = self.options.shared
+        tc.variables["CONC_WITH_UNIT_TESTS"] = False
+        tc.variables["WITH_SSL"] = self.options.with_ssl
+        tc.variables["WITH_EXTERNAL_ZLIB"] = True
+        tc.variables["INSTALL_BINDIR"] = "bin"
+        tc.variables["INSTALL_LIBDIR"] = "lib"
+        tc.variables["INSTALL_PLUGINDIR"] = os.path.join("lib", "plugin").replace("\\", "/")
+        # To install relocatable shared libs on Macos
+        tc.cache_variables["CMAKE_POLICY_DEFAULT_CMP0042"] = "NEW"
```

Comment:
```suggestion
        tc.cache_variables["CMAKE_POLICY_DEFAULT_CMP0042"] = "NEW"
        tc.cache_variables["CMAKE_POLICY_DEFAULT_CMP0077"] = "NEW"
```
For `tc.variables` to work correctly.